src/app/components/my-stats/my-stats.component.ts

```diff
@@ -0,0 +1,86 @@
+import {Component, OnInit,ChangeDetectionStrategy} from '@angular/core'
+import {Category, User} from '@app/_models'
+import {ActivatedRoute} from '@angular/router'
+import {AuthenticationService} from '@app/_services/api/authentication'
+import {ApiService} from "@app/_services/api.service"
+import {Stats} from "@app/_models/user_difficulty_stats"
+import {UserStatsService} from "@app/_services/api/user-stats.service"
+import {CategoryService} from "@app/_services/api/category.service"
+import {DifficultyService} from "@app/problems/_services/difficulty.service"
+import {Difficulty} from "@app/_models/difficulty"
+
+@Component({
+    selector: 'app-my-stats',
+    templateUrl: './my-stats.component.html',
+    styleUrls: ['./my-stats.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MyStatsComponent implements OnInit {
+    difficulties: Difficulty[] = []
+    pCategories: Category[] = []
+    user: User
+    stats: Stats
+    challengesDone: number
+    goalsDone: number
+
+    mcqNum: number
+    parsonsNum: number
+    javaNum: number
+
+    totalqDone: number
+    questionsSolved: number[] = []
+    qSolvedDifficulty: number[] = []
+
+    constructor(
+        private route: ActivatedRoute,
+        private authenticationService: AuthenticationService,
+        private apiService: ApiService,
+        private userStatsService: UserStatsService,
+        private categoryService: CategoryService,
+        private difficultyService: DifficultyService
+
+    ) {
+        this.authenticationService.currentUser.subscribe(user => this.user = user)
+    }
+
+    async ngOnInit(): Promise<void> {
+        this.categoryService.getCategories().subscribe((categories) => {
+            this.pCategories = categories.filter(c => !c.parent)
+
+            this.userStatsService.getUserStats().subscribe(stats => {
+                this.stats = stats
+                this.challengesDone = stats.challenge_stats.challenges_completed
+                this.goalsDone = stats.goal_stats.goals_completed
+
+                for( const cat of this.pCategories){
```

Comment:
The linting is off here `for (const cat of this....) {`

Reply:
I'm not getting a lint error, if I change it to 'let' instead of const it gives me a listing error.